gcloud/monitoring/group.py

```diff
@@ -0,0 +1,502 @@
+    projects.groups
+"""
+
+import re
+
+from gcloud._helpers import _name_from_project_path
+from gcloud.exceptions import NotFound
+from gcloud.monitoring._helpers import _format_timestamp
+from gcloud.monitoring.resource import Resource
+
+
+_GROUP_TEMPLATE = re.compile(r"""
+    projects/            # static prefix
+    (?P<project>[^/]+)   # initial letter, wordchars + hyphen
+    /groups/             # static midfix
+    (?P<name>[^/]+)      # initial letter, wordchars + allowed punc
+""", re.VERBOSE)
+
+
+def group_id_from_name(path, project=None):
+    """Validate a group URI path and get the group ID.
+
+    :type path: string
+    :param path: URI path for a group API request.
+
+    :type project: string or None
+    :param project: The project associated with the request. It is
+                    included for validation purposes.
+
+    :rtype: string
+    :returns: Group ID parsed from ``path``.
+    :raises: :class:`ValueError` if the ``path`` is ill-formed or if
+             the project from the ``path`` does not agree with the
+             ``project`` passed in.
+    """
+    return _name_from_project_path(path, project, _GROUP_TEMPLATE)
+
+
+def group_name_from_id(project, group_id):
+    """Build the group name given the project and group ID.
+
+    :type project: string
+    :param project: The project associated with the group.
+
+    :type group_id: string
+    :param group_id: The group ID.
+
+    :rtype: string
+    :returns: The fully qualified name of the group.
+    """
+    return 'projects/{project}/groups/{group_id}'.format(
+        project=project, group_id=group_id)
+
+
+class Group(object):
+    """A dynamic collection of monitored resources.
+
+    :type client: :class:`gcloud.monitoring.client.Client`
+    :param client: A client for operating on the metric descriptor.
+
+    :type group_id: string or None
+    :param group_id: The ID of the group.
+
+    :type display_name: string or None
+    :param display_name:
+        A user-assigned name for this group, used only for display purposes.
+
+    :type parent_id: string or None
+    :param parent_id:
+        The ID of the group's parent, if it has one.
+
+    :type filter_string: string or None
+    :param filter_string:
+        The filter string used to determine which monitored resources belong to
+        this group.
+
+    :type is_cluster: boolean
+    :param is_cluster:
+        If true, the members of this group are considered to be a cluster. The
+        system can perform additional analysis on groups that are clusters.
+    """
+    def __init__(self, client, group_id=None, display_name=None,
+                 parent_id=None, filter_string=None, is_cluster=False):
+        self.client = client
+        self._id = group_id
+        self.display_name = display_name
+        self.parent_id = parent_id
+        self.filter = filter_string
+        self.is_cluster = is_cluster
+
+        if group_id:
```

Comment:
This class allows group_id/parent_id to be empty strings to specify group names that are `None`.